api/src/main/java/io/minio/HttpRequestBody.java

```diff
@@ -71,7 +71,7 @@ public long contentLength() {
   public void writeTo(BufferedSink sink) throws IOException {
     if (data instanceof InputStream) {
       InputStream stream = (InputStream) data;
-      sink.write(Okio.source(stream), len);
+      sink.writeAll(Okio.source(stream));
```

Comment:
Could you share why `sink.Write()` is a problem?

Reply:
That is the cause of the hang @balamurugana

Reply:
For now will keep this in the PR until we get clarity upstream.